kong/templates/kong_defaults.lua

```diff
@@ -30,6 +30,8 @@ cluster_server_name = NONE
 cluster_data_plane_purge_delay = 1209600
 cluster_ocsp = off
 
+lmdb_environment_path = dbless.lmdb
```

Comment:
should we just call it `lmdb` instead?

Reply:
I am considering we may introduce other LMDB databases later, calling it `lmdb` only seems to not tell clearly what the directory is for.

In fact, I even considered remove the "lmdb" part from the `lmdb_environment_path` config and call it `dbless_database_path` instead. Does it seems to be a good idea?

Reply:
The `dbless` is little bit strange to me, as it is used on hybrid data planes too, and not only on dbless node. Yes, currently hybrid data planes are kinda dbless nodes, but lmdb really also feels like a db, just that it is not an external dependency. Also I don't like dots in directory names, as they look more like files. But well.. I have no strong feelings, it just looked a bit off to me.

Reply:
Agreed that using something that has `DB` on its name to implement something called `db-less` feels ... a bit ridicule, and inexact.

I think what differentiates the strategy presented on this PR versus PostgreSQL is **individual updates**: you can't update a single Service with the Admin API, and you can't create an individual oauth2 token. You have to change the whole thing, or none at all. So for the strategy presented on this PR I would go with something along the lines of: `static`, `monolith`, `bulk` or `unit`.

I would reserve the `lmdb` name for a strategy that makes full use of lmdb (one that allows for individual Service updates), if we ever do that.

(I considered `memory`, but lmdb apparently can write to the hdd, so that one's discarded. `read-only` is also discarded, since you can make updates, just not individual ones).

I also want to mention that the *cost* of making this change:
* It's **very** backwards-breaking, so it blocks this into 3.0 territory. It is unlikely that we would be able to include it into 2.8, but if we later decide to release 2.9 or 2.10 (which seems unlikely, but things change fast in software) it would be blocked for those too.
* Changing this in the docs could be a hassle. Not only we mention db-vs-dbless in a lot of reference pages, the Admin API differentiates all methods into `dbless-compatible` and `db-only` (or something along those lines). Plugins are also marked as `db-less` compatible. We might need to release a new version of every plugin. Old blogposts will be out of date.

That cost ... already gives me pause. We could avoid all that by just "endure the feeling of ridiculousness and inexactitude".

Reply:
Perhaps then:
```
lmdb_environment_path = dbless
```

and call it done. remove the lmdb from it, if that is easier. We could call it `config` as well, as that is what we store in there. Or have a single `lmdb` directory and under that `dbless`. I don't know.